nixos/modules/services/networking/ctrld.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ctrld;
+  settingsFormat = pkgs.formats.toml {};
+in {
+
+  options.services.ctrld = {
+    enable = mkEnableOption "ctrld service";
+
+    settings = mkOption {
+      type = settingsFormat.type;
+
+      # https://github.com/Control-D-Inc/ctrld/blob/main/README.md#default-config
+      default = {
+        listener."listener.0" = {
+          ip = "";
+          port = 0;
+          restricted = false;
+        };
+
+        network."network.0" = {
+          cidrs = [ "0.0.0.0/0" ];
+          name = "Network 0";
+        };
+
+        service = {
+          log_level = "info";
+          log_path = "";
+        };
+
+        upstream = {
+          "upstream.0" = {
+            bootstrap_ip = "76.76.2.11";
+            endpoint = "https://freedns.controld.com/p1";
+            name = "Control D - Anti-Malware";
+            timeout = 5000;
+            type = "doh";
+          };
+
+          "upstream.1" = {
+            bootstrap_ip = "76.76.2.11";
+            endpoint = "p2.freedns.controld.com";
+            name = "Control D - No Ads";
+            timeout = 3000;
+            type = "doq";
+          };
+        };
+      };
```

Comment:
These upstreams strike me as odd defaults for nixos to promote. Could this config perhaps be changed into the example? The commented link to the upstream default_config could be moved into the `description` for the user to view and reference themselves.

Reply:
The simplest config imaginable should be the default

Reply:
I think an empty config would be ideal in this case. I saw that NextDNS uses an empty config so I think it should be like this.